R/teal_lockfile.R

```diff
@@ -39,50 +39,57 @@ teal_lockfile <- function() {
     }
   }
 
-  if (!(is_in_test() || is_r_cmd_check())) {
-    old_plan <- future::plan()
-    # If there is already a parallel (non-sequential) backend, reuse it.
-    if (inherits(old_plan, "sequential")) {
-      future::plan(future::multisession, workers = 2)
-    }
+  shiny::onStop(function() file.remove(lockfile_path))
 
-    lockfile_task <- ExtendedTask$new(create_renv_lockfile)
-    lockfile_task$invoke(close = inherits(old_plan, "sequential"), lockfile_path)
-    logger::log_trace("lockfile creation invoked.")
-  }
+  # # capture.output is not needed if stdout = "|"
+  # utils::capture.output( # Needed to suppress: 'Opening fd 1' message
+  callr::r_bg(
+    func = create_renv_lockfile_2,
+    args = list(
+      lockfile_path = lockfile_path,
+      opts = options()
+    ),
+    # prints results to the console
+    stdout = "",
+    # renv setup is orchestrated by its special S3 objects: renv::settings, renv::config and renv::paths
+    #     `package` parameter include `renv` namespace inside the environment of `func = create_renv_lockfile`
+    package = "renv",
+    # callr process will use environmental variables from the main session (parent process)
+    env = Sys.getenv()
+  ) # ,
+  #   type = "message"
+  # )
+
+  logger::log_trace("lockfile creation started.")
+}
+
+create_renv_lockfile_2 <- function(lockfile_path = NULL, opts = NULL) {
```

Comment:
This is just for testing, if the callr process is working or not, and when it's stopped/killed